lib/json/schema_builder/entity.rb

```diff
@@ -44,6 +44,20 @@ def initialize(name, opts = { }, &block)
         initialize_with opts
         eval_block &block
         extract_types
+        @initialized = true
+      end
+
+      def initialized?
+        !!@initialized
+      end
+
+      def reinitialize
+      end
+
+      def extend(child_name, &block)
+        child = children.find { |c| c.name == child_name.to_sym }
```

Comment:
@parrish I think it would be useful to extract this into its own method. 

```ruby
object_with_child.child(:type) #=> ObjectBuilder<:type> 
```
Not sure about the name yet, but this way you could pass children easily to other functions as well.

Reply:
Maybe it would be enough to just return the child from `extend` if `!block_given?`